MusicApp.SongService.Application/CQRS/Commands/UpdateSong/UpdateSongCommandHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using FluentValidation;
+using MediatR;
+using MusicApp.SongService.Application.Repositories;
+using MusicApp.SongService.Application.Services;
+using MusicApp.SongService.Domain.Entities;
+using MusicApp.SongService.Domain.Exceptions;
+
+namespace MusicApp.SongService.Application.CQRS.Commands.UpdateSong;
+
+public class UpdateSongCommandHandler : IRequestHandler<UpdateSongCommand, Song>
+{
+    private readonly ISongRepository _repository;
+    private readonly ArtistService _artistService;
+    private readonly UpdateSongCommandValidator _validator;
+
+    public UpdateSongCommandHandler(ISongRepository repository, ArtistService artistService)
+    {
+        _repository = repository;
+        _artistService = artistService;
+        _validator = new();
+    }
+
+    public async Task<Song> Handle(UpdateSongCommand request, CancellationToken cancellationToken)
+    {
+        await _validator.ValidateAndThrowAsync(request, cancellationToken);
```

Comment:
you don't need to do validation here anymore. MediatR's behavior pipeline will handle the validation of the request before it reaches the Handle method